### PR TITLE
e2e: stabilize and enable numaresourcesscheduler test-suite

### PR DESCRIPTION
**Pull request by Tal-or**
When the numaresourcesscheduler was implemented initially, we thought it was going to stay with us for a very short time.
This assumption turned out to be wrong, so it's worth enabling the e2e tests for numaresourcesscheduler by default in order to avoid future regression.

Signed-off-by: Talor Itzhak <titzhak@redhat.com>

### CODE REVIEW

**Comment by Tal-or** [comment]
Depend on: https://github.com/openshift-kni/numaresources-operator/pull/121 or https://github.com/openshift-kni/numaresources-operator/pull/122

**Comment by fromanirh** [comment]
I'm just wondering if we should run on a separate lane. Probably not, but worth discussing it anyway.
Another thing worth discussing if we should enable this on prow only, to not overtax kind.
Ideally, we would always run this lane, but the low resources on the gh actions lane is a concern and I'd like to save resources for the most important tests.

**Comment by fromanirh** [comment]
> Depend on: #121 or #122

I'd like to have both :)

**Comment by Tal-or** [comment]
> I'm just wondering if we should run on a separate lane. Probably not, but worth discussing it anyway. Another thing worth discussing if we should enable this on prow only, to not overtax kind. Ideally, we would always run this lane, but the low resources on the gh actions lane is a concern and I'd like to save resources for the most important tests.

I'm not sure that it will work without the installation of the NUMAresourcesOperator CR, because in the future (or even now, I need to check) we rely on the info provided us by the NRT objects which in turn rely on RTE to export it.

**Comment by fromanirh** [comment]
will review ASAP

File: Makefile [hunk 1]
```diff
@@ -120,7 +120,7 @@ test-unit: envtest
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test ./controllers/... ./pkg/... ./rte/pkg/...
 
 test-e2e: build-e2e-all
-	hack/run-test-e2e.sh
+	ENABLE_SCHED_TESTS=true hack/run-test-e2e.sh
```
Comment (fromanirh) [inline Makefile:123]: ok for now, let's think about creating a separate lane

**Review by fromanirh — COMMENTED** [review]
/lgtm
/approve
we can discuss possible further improvements later

File: Makefile (hunk 1 above)
Comment (Tal-or) [reply Makefile:123]: The problem with a separate lane is limiting the sched tests to be independent of NRT objects, so we can test the scheduler installation but nothing more.

If we want to have access to NRT objects, we need the NROP CR installation which is exactly what we're getting here.

Comment (fromanirh) [reply Makefile:123]: fair point indeed. Let's keep it here for the time being.

**Comment by fromanirh** [comment]
too many failures in a row are suspicious now. Let's check what's going on.

**Comment by fromanirh** [comment]
nope, failures seem real.

**Comment by Tal-or** [comment]
/retest

Test failed because of:

```• Failure [0.763 seconds]
with a running cluster with all the components
/go/src/github.com/openshift-kni/numaresources-operator/test/e2e/rte/rte_test.go:52
  when [config][rte] NRO CR configured with LogLevel
  /go/src/github.com/openshift-kni/numaresources-operator/test/e2e/rte/rte_test.go:75
    can modify the LogLevel in NRO CR and klog under RTE container should change respectively [It]
    /go/src/github.com/openshift-kni/numaresources-operator/test/e2e/rte/rte_test.go:93
    LogLevel Trace doesn't match the existing --v flag under &Container{Name:resource-topology-exporter,Image:quay.io/openshift-kni/numaresources-operator:4.10.999-snapshot,Command:[/bin/resource-topology-exporter],Args:[--sleep-interval=10s --sysfs=/host-sys --podresources-socket=unix:///host-podresources-socket/kubelet.sock --notify-file=/host-run-rte/notify --topology-manager-policy=single-numa-node --exit-on-conf-change --v=4],WorkingDir:,Ports:[]ContainerPort{ContainerPort{Name:metrics-port,HostPort:0,ContainerPort:2112,Protocol:TCP,HostIP:,},},Env:[]EnvVar{EnvVar{Name:NODE_NAME,Value:,ValueFrom:&EnvVarSource{FieldRef:&ObjectFieldSelector{APIVersion:v1,FieldPath:spec.nodeName,},ResourceFieldRef:nil,ConfigMapKeyRef:nil,SecretKeyRef:nil,},},EnvVar{Name:REFERENCE_NAMESPACE,Value:,ValueFrom:&EnvVarSource{FieldRef:&ObjectFieldSelector{APIVersion:v1,FieldPath:metadata.namespace,},ResourceFieldRef:nil,ConfigMapKeyRef:nil,SecretKeyRef:nil,},},EnvVar{Name:REFERENCE_POD_NAME,Value:,ValueFrom:&EnvVarSource{FieldRef:&ObjectFieldSelector{APIVersion:v1,FieldPath:metadata.name,},ResourceFieldRef:nil,ConfigMapKeyRef:nil,SecretKeyRef:nil,},},EnvVar{Name:REFERENCE_CONTAINER_NAME,Value:shared-pool-container,ValueFrom:nil,},EnvVar{Name:METRICS_PORT,Value:2112,ValueFrom:nil,},},Resources:ResourceRequirements{Limits:ResourceList{},Requests:ResourceList{},},VolumeMounts:[]VolumeMount{VolumeMount{Name:host-sys,ReadOnly:true,MountPath:/host-sys,SubPath:,MountPropagation:nil,SubPathExpr:,},VolumeMount{Name:host-podresources-socket,ReadOnly:false,MountPath:/host-podresources-socket/kubelet.sock,SubPath:,MountPropagation:nil,SubPathExpr:,},VolumeMount{Name:host-run-rte,ReadOnly:false,MountPath:/host-run-rte,SubPath:,MountPropagation:nil,SubPathExpr:,},VolumeMount{Name:rte-config-volume,ReadOnly:false,MountPath:/etc/resource-topology-exporter/,SubPath:,MountPropagation:nil,SubPathExpr:,},},LivenessProbe:nil,ReadinessProbe:nil,Lifecycle:nil,TerminationMessagePath:/dev/termination-log,ImagePullPolicy:IfNotPresent,SecurityContext:&SecurityContext{Capabilities:nil,Privileged:nil,SELinuxOptions:&SELinuxOptions{User:,Role:,Type:rte.process,Level:s0,},RunAsUser:*0,RunAsNonRoot:nil,ReadOnlyRootFilesystem:nil,AllowPrivilegeEscalation:nil,RunAsGroup:*0,ProcMount:nil,WindowsOptions:nil,SeccompProfile:nil,},Stdin:false,StdinOnce:false,TTY:false,EnvFrom:[]EnvFromSource{},TerminationMessagePolicy:File,VolumeDevices:[]VolumeDevice{},StartupProbe:nil,} container
      
    Expected
        <bool>: false
    to be true
    /go/src/github.com/openshift-kni/numaresources-operator/test/e2e/rte/rte_test.go:110```
    
 https://github.com/openshift-kni/numaresources-operator/pull/133 tries to address that

**Comment by Tal-or** [comment]
/retest
infra issue

**Comment by Tal-or** [comment]
/retest
failures are not related to our code

**Comment by Tal-or** [comment]
rebase after https://github.com/openshift-kni/numaresources-operator/pull/133 got merged

**Comment by fromanirh** [comment]
```
msg=Cluster operator etcd Degraded is True with ClusterMemberController_Error::DefragController_Error::EtcdMembers_UnhealthyMembers::StaticPods_Error: ClusterMemberControllerDegraded: unhealthy members found during reconciling members
```
again. Worth investigating and reporting to infra folks.

**Review by fromanirh — COMMENTED** [review]
/lgtm
/approve

**Comment by Tal-or** [comment]
rebase again on top of some fixes that should stabilize the CI

**Review by fromanirh — COMMENTED** [review]
/lgtm
/approve

**Comment by Tal-or** [comment]
Additional fix here:
https://github.com/openshift-kni/numaresources-operator/pull/123/commits/7a5026a20e3687093185bf4d798ffd6629339d52

**Comment by fromanirh** [comment]
possibly merged too early?

**Comment by Tal-or** [comment]
> possibly merged too early?

That's OK the MCP fix we discussed yesterday, is good in general. it's not related directly for this PR

**Pull request merged**